builds/e2e/nested-e2e.yaml

```diff
@@ -107,25 +107,50 @@ stages:
           IotHubConnectionString: $(ConnectivityNestedEdge-ConnectionString)
 
   - job:  Clean_up     
-    dependsOn: SetupVM_and_RunTest_level3
+    dependsOn:
+      - SetupVM_level5 
+      - SetupVM_level4
+      - SetupVM_and_RunTest_level3
     condition: always()
     displayName: Clean up identities and unlock agents
     timeoutInMinutes: 1    
     variables:    
-      deviceLvl5Name: $[ dependencies.SetupVM_level5.outputs['deployIoTEdge.deviceName'] ]
-      deviceLvl5ConnectionString: $[ dependencies.SetupVM_level5.outputs['createIdentity.connectionString'] ]     
-      deviceLvl4Name: $[ dependencies.SetupVM_level4.outputs['deployIoTEdge.deviceName'] ]
-      deviceLvl4ConnectionString: $[ dependencies.SetupVM_level4.outputs['createIdentity.connectionString'] ] 
+      deviceLvl5DeviceId: $[ dependencies.SetupVM_level5.outputs['createIdentity.parentDeviceId'] ] 
+      deviceLvl4DeviceId: $[ dependencies.SetupVM_level4.outputs['createIdentity.parentDeviceId'] ] 
+      deviceLvl5IoTHubName: $[ dependencies.SetupVM_level5.outputs['createIdentity.iotHubName'] ] 
+      deviceLvl4IoTHubName: $[ dependencies.SetupVM_level4.outputs['createIdentity.iotHubName'] ]
+      deviceLvl5AgentName: $[ dependencies.SetupVM_level5.outputs['lock_test_agent.agentName'] ]         
+      deviceLvl4AgentName: $[ dependencies.SetupVM_level4.outputs['lock_test_agent.agentName'] ]       
+      deviceLvl3AgentName: $[ dependencies.SetupVM_level3.outputs['lock_test_agent.agentName'] ]  
     pool:
       name: $(pool.name)
       demands:
         - agent-group -equals $(agent.group)
         - Agent.OS -equals Linux
         - Agent.OSArchitecture -equals X64
         - run-idle -equals true 
-        - status -equals unlocked_$(Build.BuildId)  
-    strategy:
-      parallel: 3          
+        - status -equals unlocked_$(Build.BuildId)          
     steps:
-      - template: templates/nested-get-secrets.yaml     
-      - template: templates/unlock-test-agent.yaml      
+      - template: templates/nested-get-secrets.yaml  
+      - template: templates/unlock-test-agent.yaml
```

Comment:
I think it would be cleaner here instead of having a *.yaml with parameter input, you do a bash function that takes parameter array input. This would prevent the duplication of code.

Reply:
Maybe, there is only 3 lvls. It is pretty subjective at that point. It would still be 2 scripts instead of 5.
The major downside is that you don't get separate red when there is a failure